src/main/java/org/apache/sling/graphql/api/SelectedField.java

```diff
@@ -39,6 +39,34 @@ public interface SelectedField {
     @Nullable
     String getName();
 
+    /** @return the simple qualified name of the selected field **/
+    @Nullable
+    String getQualifiedName();
+
+    /**
+     * @return the fully qualified name of the item
+     */
+    @Nullable
+    String getFullyQualifiedName();
+
+    /** @return level of the selected field within the query **/
+    int getLevel();
+
+    /** @return whether the field is conditionally present **/
+    boolean isConditional();
+
+    /**
+     * @return the alias of the selected field or null if not alias was used
+     */
+    String getAlias();
+
+    /**
+     * The result key is either the field query alias OR the field name in that preference order
+     *
+     * @return the result key of the selected field
+     */
+    String getResultKey();
+
```

Comment:
It is Jetbrains null annotations nowadays (https://sling.apache.org/documentation/development/null-analysis.html) :-)

Reply:
No, I cannot because the source of the values does not provide any annotations or any other indication of the expectation. I will mark the all Nullable for now